component/newstorage/formattedstore/formattedstore.go

```diff
@@ -0,0 +1,395 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package formattedstore
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/newstorage"
+)
+
+const (
+	expressionTagNameOnlyLength     = 1
+	expressionTagNameAndValueLength = 2
+
+	failFormat               = `failed to format %s "%s": %w`
+	failFormatTags           = "failed to format tags: %w"
+	failDeformat             = `failed to deformat %s "%s" returned from the underlying store: %w`
+	failQueryUnderlyingStore = "failed to query underlying store: %w"
+)
+
+var errInvalidQueryExpressionFormat = errors.New("invalid expression format. " +
+	"it must be in the following format: TagName:TagValue")
+
+// Formatter represents a type that can convert data between two formats.
+type Formatter interface {
+	FormatKey(key string) (formattedKey string, err error)
```

Comment:
Reason for multiple format functions: the EDV formatter implementation will have different formats for data (encrypted "format") vs. keys/tags (MAC values). Other formatters might too.